### DISCUSSION TITLE
Major Overhaul

### DISCUSSION THREAD

**Discussion by gilesknap** [gilesknap/gphotos-sync#349] [Announcements]
## Recent Work 

This release has one very important bug fix. 
- Switch to supported OAuth authentication flow. #341 -  Thanks @marcocaberletti  !

Having this fix has inspired me to spend some effort on this project to make it more easily maintainable in future.

I have totally refreshed the developer workflow and documentation as follows:
- completely new sphinx based documentation hosted on GitHub Pages
- switch CI to Github Actions
- add multi-arch container build to CI
- add devcontainer for vscode 
- container image hosted on GitHub Container Registry
- remove pipenv dependency and use pip only (with lock semantics in the CI)
- add multiple source checks to CI and dev time
  - black
  - flake8
  - mypy
  - isort
  - yaml checks
- implement git pre-commit checks for some of the above
- use tox for parallel execution of all the above

There are also multiple contributions from others listed below that have accumulated because it's been so long since the last release.
Sorry about that. I intend to pay more attention to this project now that it is in a tidy state.

## What's Changed
* Update README.rst by @xyzulu in https://github.com/gilesknap/gphotos-sync/pull/235
* clearly state is not for uploading (closes #245) by @maphew in https://github.com/gilesknap/gphotos-sync/pull/246
* handle exception 'OSError: [Errno 36] File name too long' by @ericjung in https://github.com/gilesknap/gphotos-sync/pull/249
* Add support for selecting multiple albums by regex by @rogerdahl in https://github.com/gilesknap/gphotos-sync/pull/251
* doc: enhance windows install with wsl2 by @cboudereau in https://github.com/gilesknap/gphotos-sync/pull/263
* Typo paramter by @gothicx in https://github.com/gilesknap/gphotos-sync/pull/260
* Make docker image lighter with alpine and multi-stage build by @esabouraud in https://github.com/gilesknap/gphotos-sync/pull/265
* Fixed failed logging when filename was UTF-8 encoded by @jukka5 in https://github.com/gilesknap/gphotos-sync/pull/272
* Fix to symlinks on NTFS by @jukka5 in https://github.com/gilesknap/gphotos-sync/pull/273
* Update description of Google issues by @benmccann in https://github.com/gilesknap/gphotos-sync/pull/277
* Apply max_filename to album links as well by @lutzky in https://github.com/gilesknap/gphotos-sync/pull/287
* multiarch image by @pheel in https://github.com/gilesknap/gphotos-sync/pull/297
* change link name in readme by @brianrook in https://github.com/gilesknap/gphotos-sync/pull/296
* Bump pyyaml from 5.3.1 to 5.4 by @dependabot in https://github.com/gilesknap/gphotos-sync/pull/307
* Output exception details when failing to link by @lutzky in https://github.com/gilesknap/gphotos-sync/pull/309
* Demote duplicate link log message to info by @lutzky in https://github.com/gilesknap/gphotos-sync/pull/310
* Add docs/google-project-setup.rst and link to it from README.rst. by @dougthor42 in https://github.com/gilesknap/gphotos-sync/pull/316
* Bump urllib3 from 1.26.4 to 1.26.5 by @dependabot in https://github.com/gilesknap/gphotos-sync/pull/315
* Add note about photo compression and link to relevant issue by @gurchik in https://github.com/gilesknap/gphotos-sync/pull/322
* Add a tip for naming the OAuth app name by @gurchik in https://github.com/gilesknap/gphotos-sync/pull/321
* Bump pywin32 from 227 to 301 by @dependabot in https://github.com/gilesknap/gphotos-sync/pull/323
* Added configurable album path formatting by @BriceSchaffner in https://github.com/gilesknap/gphotos-sync/pull/329
* Allow the inversion of the file order in albums. by @exi in https://github.com/gilesknap/gphotos-sync/pull/330
* Replace Google authentication strategy by @marcocaberletti in https://github.com/gilesknap/gphotos-sync/pull/342
* Fixes for new auth flow by @gilesknap in https://github.com/gilesknap/gphotos-sync/pull/343
* Major Restructure by @gilesknap in https://github.com/gilesknap/gphotos-sync/pull/348

## New Contributors
* @xyzulu made their first contribution in https://github.com/gilesknap/gphotos-sync/pull/235
* @maphew made their first contribution in https://github.com/gilesknap/gphotos-sync/pull/246
* @ericjung made their first contribution in https://github.com/gilesknap/gphotos-sync/pull/249
* @rogerdahl made their first contribution in https://github.com/gilesknap/gphotos-sync/pull/251
* @gothicx made their first contribution in https://github.com/gilesknap/gphotos-sync/pull/260
* @esabouraud made their first contribution in https://github.com/gilesknap/gphotos-sync/pull/265
* @jukka5 made their first contribution in https://github.com/gilesknap/gphotos-sync/pull/272
* @benmccann made their first contribution in https://github.com/gilesknap/gphotos-sync/pull/277
* @lutzky made their first contribution in https://github.com/gilesknap/gphotos-sync/pull/287
* @pheel made their first contribution in https://github.com/gilesknap/gphotos-sync/pull/297
* @brianrook made their first contribution in https://github.com/gilesknap/gphotos-sync/pull/296
* @dependabot made their first contribution in https://github.com/gilesknap/gphotos-sync/pull/307
* @dougthor42 made their first contribution in https://github.com/gilesknap/gphotos-sync/pull/316
* @gurchik made their first contribution in https://github.com/gilesknap/gphotos-sync/pull/322
* @BriceSchaffner made their first contribution in https://github.com/gilesknap/gphotos-sync/pull/329
* @exi made their first contribution in https://github.com/gilesknap/gphotos-sync/pull/330
* @marcocaberletti made their first contribution in https://github.com/gilesknap/gphotos-sync/pull/342

**Full Changelog**: https://github.com/gilesknap/gphotos-sync/compare/2.14.2...3.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/gilesknap/gphotos-sync/releases/tag/3.0.0'>Major Overhaul</a>.</em>

**Comment by mrzoops** [comment]
No zip file for this release?

**Reply by gilesknap** [reply]
I am seeing a source zip at https://github.com/gilesknap/gphotos-sync/releases/tag/3.0.0 
Is that what you mean?
